### PR TITLE
Bug fix of MF face eval for hanging nodes and non-standard orientation

### PR DESCRIPTION
**Pull request by kronbichler**
While working on an example with hanging nodes along a face with non-standard orientation, I observed that `FEFaceEvaluation` had two bugs:
- The first bug, introduced last week by #13056, contained an unnecessary loop over components (not covered by tests as we only looked into the scalar case in detail for the orientations).
- The second and more serious bug is that we did not account for the right sub-face in case of hanging nodes at the interface of orientations in the setup of the matrix-free faces. This bug has been present since we merged the MF face integrals in 2018 but it seems that we never looked closely. It seems that I wrongly assumed that https://github.com/dealii/dealii/blob/5cc500778d1ecad7c3464e18c94383453bb198fb/source/grid/tria_accessor.cc#L2350-L2352 would give us the right info, which it apparently does not (by comparing to a matrix-based computation).

### CODE REVIEW

File: include/deal.II/matrix_free/face_setup_internal.h [hunk 1]
```diff
@@ -1041,6 +1042,20 @@ namespace internal
...
         info.face_orientation = exterior_face_orientation;
+
+      // make sure to select correct subface index in case of non-standard
+      // orientation of the coarser neighbor face
+      if (cell->level() > neighbor->level() && exterior_face_orientation > 0)
+        {
+          Table<2, unsigned int> orientation =
+            ShapeInfo<double>::compute_orientation_table(2);
+          std::array<unsigned int, 8> inverted_orientations{
+            {0, 1, 2, 3, 6, 5, 4, 7}};
```
Comment (peterrum) [inline include/deal.II/matrix_free/face_setup_internal.h:1053]: I am wondering if we should have this at a central place?

File: include/deal.II/matrix_free/face_setup_internal.h [hunk 2]
```diff
@@ -1041,6 +1042,20 @@ namespace internal
         }
       else
         info.face_orientation = exterior_face_orientation;
+
+      // make sure to select correct subface index in case of non-standard
+      // orientation of the coarser neighbor face
+      if (cell->level() > neighbor->level() && exterior_face_orientation > 0)
+        {
+          Table<2, unsigned int> orientation =
```
Comment (peterrum) [inline include/deal.II/matrix_free/face_setup_internal.h:1050]:
```suggestion
          const Table<2, unsigned int> orientation =
```

File: include/deal.II/matrix_free/face_setup_internal.h [hunk 3]
```diff
@@ -1041,6 +1042,20 @@ namespace internal
...
+            ShapeInfo<double>::compute_orientation_table(2);
+          std::array<unsigned int, 8> inverted_orientations{
```
Comment (peterrum) [inline include/deal.II/matrix_free/face_setup_internal.h:1052]:
```suggestion
          const std::array<unsigned int, 8> inverted_orientations{
```

**Review by peterrum — APPROVED** [review]

File: include/deal.II/matrix_free/face_setup_internal.h (hunk 1 above)
Comment (kronbichler) [reply include/deal.II/matrix_free/face_setup_internal.h:1053]: I think so, but I won't have time today and tomorrow to fix it. Maybe we can do that in a follow-up PR?

**Review by peterrum — APPROVED** [review]

**Pull request merged**